### ISSUE TITLE
Vulkan swapChainImages and swapChainFrameBuffers not being properly deleted when SwapChain is recreated.

### ISSUE THREAD

**Issue by hexsocean** [LukasBanana/LLGL#76]
I have identified bugs at
https://github.com/LukasBanana/LLGL/blob/master/sources/Renderer/Vulkan/VKRenderContext.cpp#L411-L416
and
https://github.com/LukasBanana/LLGL/blob/master/sources/Renderer/Vulkan/VKRenderContext.cpp#L465-L470

the outdated VKPtr<VkImageView> and VKPtr<VkFramebuffer> don't get deleted, instead a new object is made every time the SwapChain is recreated; this causes it not to be deleted on the GPU.

I think it should check if one of the objects exists, if it doesn't, it should create one; then it can recreate it. Like this:
```cpp
	if (!swapChainImageViews_[i]) {
		swapChainImageViews_[i] = VKPtr<VkImageView>{device_, vkDestroyImageView};
	}

	auto result = vkCreateImageView(device_, &createInfo, nullptr, wapChainImageViews_[i].ReleaseAndGetAddressOf());
	VKThrowIfFailed(result, "failed to create Vulkan swap-chain image view");
```
```cpp
	if (!swapChainFramebuffers_[i]) {
		swapChainFramebuffers_[i] = VKPtr<VkFramebuffer>{device_, vkDestroyFramebuffer};
	}

	auto result = vkCreateFramebuffer(device_, &createInfo, nullptr, swapChainFramebuffers_[i].ReleaseAndGetAddressOf());
	VKThrowIfFailed(result, "failed to create Vulkan swap-chain framebuffer");
```

**Comment by LukasBanana** [comment]
Thank you. I can take a look next week.

**Comment by LukasBanana** [comment]
`swapChainFramebuffers_[]` is an array of `VKPtr` which should handle the deletion in the move operator, but I guess that operator is wrong in not calling the assigned deleter:
https://github.com/LukasBanana/LLGL/blob/master/sources/Renderer/Vulkan/VKPtr.h#L131

**Comment by hexsocean** [comment]
The issue is fixed after that change.

**Comment by LukasBanana** [comment]
Thanks for pointing out this issue. Fixed with [9dc06e5](https://github.com/LukasBanana/LLGL/commit/9dc06e500b3f5e6ab92c007b5301ae9eb28dfe47).
